lib/cdo/delete_accounts_helper.rb

```diff
@@ -218,11 +227,33 @@ def purge_teacher_feedbacks(user_id)
     end
   end
 
+  def check_safety_constraints(user)
+    assert_constraint !user.facilitator?,
+      'Automated purging of facilitator accounts is not supported at this time.'
+    assert_constraint RegionalPartner.with_deleted.where(contact_id: user.id).empty?,
```

Comment:
The better way to do this would be `user.permission?(UserPermission::FACILITATOR) || user.permission?(UserPermission::WORKSHOP_ORGANIZER) || user.permission?(UserPermission::PROGRAM_MANAGER)`

Reply:
Can you explain that?  I don't understand the relationship between regional partners, workshop organizers, and program managers.  Are these all distinct types of accounts we would not be okay with auto-purging?  Is it possible for a `RegionalPartner.contact` to be a user account that does not have one of these permissions?

Reply:
Got an overview from Andrew this morning, so I see what you're saying now.  It sounds like we don't want to auto-purge any user meeting one or more of the following conditions:

- Has Facilitator, Workshop Organizer or Program Manager permission.
- Is referenced by `RegionalPartner#contact` (although this is being phased out)
- Is referenced by `RegionalPartner#program_managers`